code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/LegacyNotificationBuilderTest.kt

```diff
@@ -0,0 +1,161 @@
+import junit.framework.TestCase.assertNull
+import org.junit.Assert.assertArrayEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class LegacyNotificationBuilderTest {
+
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var context: Context
+    private lateinit var dataMap: MutableMap<String, String>
+    private lateinit var mockBundle: Bundle
+
+    @Before
+    fun setUp() {
+        context = RuntimeEnvironment.getApplication()
+        trackerActivityClass = DummyActivity::class.java
+        dataMap = MockAEPPushTemplateDataProvider.getMockedAEPDataMapWithAllKeys()
+        mockBundle = MockAEPPushTemplateDataProvider.getMockedAEPBundleWithAllKeys()
+    }
+
+    @Test
+    fun `verify construct should map valid BasicPushTemplate data to notification data`() {
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val notificationBuilder =
+            LegacyNotificationBuilder.construct(context, pushTemplate, trackerActivityClass)
+        val pendingIntent = notificationBuilder.build().contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+        assertEquals(pushTemplate.ticker, notificationBuilder.build().tickerText)
+        assertEquals(
+            pushTemplate.title,
+            notificationBuilder.build().extras.getString(NotificationCompat.EXTRA_TITLE)
+        )
+        assertEquals(
+            pushTemplate.body,
+            notificationBuilder.build().extras.getString(NotificationCompat.EXTRA_TEXT)
+        )
+        assertEquals(
+            pushTemplate.body,
+            notificationBuilder.build().extras.getString(NotificationCompat.EXTRA_TEXT)
+        )
+        assertEquals(pushTemplate.channelId, notificationBuilder.build().channelId)
+        assertNotNull(notificationBuilder.build().smallIcon)
+        assertEquals(
+            pushTemplate.actionButtonsList?.map { it.label },
+            notificationBuilder.build().actions.map { it.title }
+        )
+        assertNotNull(notificationBuilder.build().deleteIntent)
+        assertEquals(
+            pushTemplate.actionUri,
+            intent.getStringExtra(PushTemplateConstants.Tracking.TrackingKeys.ACTION_URI)
+        )
+        assertEquals(
+            pushTemplate.tag,
+            intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.TAG)
+        )
+        assertEquals(
+            pushTemplate.isNotificationSticky,
+            intent.getBooleanExtra(PushTemplateConstants.PushPayloadKeys.STICKY, false)
+        )
+    }
+
```

Comment:
For these test cases below which verify specific condtions, can you also verify that the other attributes of the notification are set as desired?

Reply:
have added a test case to verify sticky notification values but apart from that it is only being used to create a pending intent

Reply:
Sounds good.  
My suggestion was more on the lines of asserting that changing or modifying data for the test cases below does not affect other attributes. This approach will ensure that the test case breaks if any linked breaking changes (channel id somehow modifying priority or vibration status) are done later during development forcing the developer to add or modify tests per their new requirements.

```
fun `construct should set default channel ID if pushTemplate channelId is null`() {
   // 1.  Remove key from mock data
   dataMap.removeKeysFromMap(PushTemplateConstants.PushPayloadKeys.CHANNEL_ID)
   
  // 2.  verify outcome of that specific key
  assertEquals(DEFAULT_CHANNEL_ID, notification.channelId)

   // 3. Verify that rest of the notification attributes are same as the ones derived from mock data (i.e and that this does not affect anything else).
 
}
```

Reply:
sure, will update the test case for Default channel ID to verify other notification attributes as well